src/trigger.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Provides an interface for an object that can generate an event.
+
+use std::result::Result;
+
+/// Abstraction for a simple, push-button like interrupt mechanism.
```

Comment:
Besides what it is, I think it would be important to add some details about the "why". Why did you add this trait? What are you trying to abstract? When are developers of this crate (or users of this interface) suppose to use it?

Reply:
I added more details here, hope it's better.